src/models/classes/container.js

```diff
@@ -36,19 +36,28 @@ export default class Container extends StorageCore
      *
      * @override
      */
-    removeItem(product)
+    removeItem(product, perishedProduct)
     {
         this.items = this.items.filter(function (item) {
             if (product.name != item.name) {
                 return item;
             }
 
-            const removedQuantity = Math.min(product.values.quantity, item.values.quantity);
+            if (product.values.isPerishable) {
+                const removedQuantity = perishedProduct.values.quantity;
 
-            product.values.quantity -= removedQuantity;
-            item.values.quantity -= removedQuantity;
+                product.values.quantity -= removedQuantity;
+                item.values.quantity -= removedQuantity;
 
-            return item.values.quantity;
+                return item.values.quantity;
```

Comment:
You should return `product.values.quantity`

Don't return here, use the one at the end of the function.